app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/Event.java

```diff
@@ -33,13 +32,23 @@
      * An image representing the event, may be null
      */
     private Bitmap image;
+    /**
+     * The location of the event
+     */
+    private EventLocation location;
+    /**
+     * A particular place into the event
+     */
+    private List<Spot> spotList;
 
-    public Event(int id, String name, String description, EventOrganizer organizer, Bitmap image) {
+    public Event(int id, String name, String description, EventOrganizer organizer, Bitmap image, EventLocation location, List<Spot> spotList) {
         this.id = id;
         this.name = name;
         this.description = description;
         this.organizer = organizer;
         this.image = image;
+        this.location = location;
+        this.spotList = new ArrayList<>(spotList);
```

Comment:
Is it really necessary to copy spotList into new ArrayList ?

Reply:
I believe that we have to deep copy here, haven't we ?

Reply:
If I'm right, the array list constructor is actually not doing a deep copy but a shallow one( https://stackoverflow.com/questions/47862684/is-the-copy-constructor-of-javas-arraylist-shallow-copy-or-deep-copy)
As it's always a performance trade-off doing a deep or shallow copy, I'm not sure it's critical here to do a deep copy in this case. But if you really wanted to do a deep copy, I think you should do it explicitly.